### PR TITLE
OCPBUGS#17371: Wipe disks before deploying LVMS on private CI environment

### PR DESCRIPTION
**Pull request by sr1kar99**
Version(s): 4.12+

Issue: [OCPBUGS-17371](https://issues.redhat.com/browse/OCPBUGS-17371)

Link to docs preview:
https://69930--ocpdocs-pr.netlify.app/openshift-enterprise/latest/storage/persistent_storage/persistent_storage_local/persistent-storage-using-lvms#lvms-deployment-requirements-for-sno-ran_logical-volume-manager-storage

QE review:
- [x] QE has approved this change.

### CODE REVIEW

**Comment by sr1kar99** [comment]
@arthur-r-oliveira 
Can you please review this PR?
Thanks.

**Comment by arthur-r-oliveira** [comment]
@sr1kar99 thanks for the heads-up. lgtm!

**Comment by sr1kar99** [comment]
@radeore 
Can you please review this PR?
Thanks.

**Comment by radeore** [comment]
`wipefs -a <device>` command alone will not help removing disks with VGs, LVs, etc. as it throws `device or resource busy `error. Before performing wipefs, need to run `dmsetup remove <vgName/lvName>`.

@suleymanakbas91 Should we mention `dmsetup remove` in user docs, considering permanent data loss?

**Comment by suleymanakbas91** [comment]
@radeore you're right. `wipefs -a` is unfortunately not a magic command to wipe a disk in all situations. If the disk was used in an LVM system before, you also need to run `dmsetup remove`. If the disk was mounted, you need to `unmount` it. If the disk was used as a swap device, you need to `unswap` it. If the disk was used in a RAID array, you need to remove it from the RAID. This list goes on and on. So, I'd try to refrain from using any commands in the docs about wiping the disk. We can mention that the disk should not be in use and should be wiped, without giving any specific commands as we cannot know the use case the user is utilizing the disk for.

**Comment by sr1kar99** [comment]
@radeore @suleymanakbas91 
I have updated the content based on the discussion. 
Could you PTAL, and suggest updates, if any.
Thanks.

File: modules/deploying-lvms-on-sno-cluster.adoc [hunk 1]
```diff
@@ -25,11 +25,17 @@ You can deploy {lvms} using one of the following:
...
+Before deploying {lvms} on {sno} clusters, ensure that the following requirements are met:
 
 * You have installed {rh-rhacm-first} on an {product-title} cluster.
 * Every managed cluster has dedicated disks that are used to provision storage.
 
+Before deploying {lvms} on a private CI environment, where virutal machines and storage devices can be reused, ensure that you have wiped the disks that are not in use. 
+[NOTE]
+====
+You can wipe only the disks that are not in use.
+====
```
Comment (agantony) [inline modules/deploying-lvms-on-sno-cluster.adoc:37]:
```suggestion
Before deploying {lvms} in a private CI environment where you can reuse virtual machines and storage devices, ensure that you have deleted the disks that are not in use.

[NOTE]
====
You can only delete the disks that are not in use.
====
```

**Review by agantony — COMMENTED** [review]
Added a suggestion for your consideration; overall lgtm!

File: modules/deploying-lvms-on-sno-cluster.adoc (hunk 1 above)
Comment (sr1kar99) [reply modules/deploying-lvms-on-sno-cluster.adoc:37]: @agantony 
Thank you for the review. Just one query on the comment-

When we say "You can delete only the disks that are not in use" - the limitation is on the type of devices that can be deleted.
In case of  "You can only delete the disks that are not in use" - the limitation is the action of deletion.

Or to avoid confusion we can rephrase it to - "You cannot delete the disks that are in use"
Any comments on this approach?

@suleymanakbas91 Could you please confirm if there is a technical difference between "deleting" and "wiping" the disks.

Comment (agantony) [reply modules/deploying-lvms-on-sno-cluster.adoc:37]: "You cannot delete the disks that are in use" --> This SGTM!

Comment (suleymanakbas91) [reply modules/deploying-lvms-on-sno-cluster.adoc]: `wipe` is the common usage, so I'd stick to that.

File: modules/deploying-lvms-on-sno-cluster.adoc [hunk 2]
```diff
@@ -25,11 +25,17 @@ You can deploy {lvms} using one of the following:
 [id="lvms-deployment-requirements-for-sno-ran_{context}"]
 == Requirements
 
-Before you begin deploying {lvms}, ensure that the following requirements are met:
+Before deploying {lvms} on {sno} clusters, ensure that the following requirements are met:
 
 * You have installed {rh-rhacm-first} on an {product-title} cluster.
 * Every managed cluster has dedicated disks that are used to provision storage.
 
+Before deploying {lvms} on a private CI environment, where virutal machines and storage devices can be reused, ensure that you have wiped the disks that are not in use.
```
Comment (bergerhoffer) [inline modules/deploying-lvms-on-sno-cluster.adoc:33]: Typo "virutal". Plus @agantony had mentioned an improvement that I wasn't sure if you saw/considered. I think it's a good suggestion that puts it more in active voice.

Posting here to fix the typo and use his suggestion, except putting it back to "wiped".

```suggestion
Before deploying {lvms} in a private CI environment where you can reuse virtual machines and storage devices, ensure that you have wiped the disks that are not in use.
```

**Review by bergerhoffer — COMMENTED** [review]
A typo plus a wording suggestion to consider, before we can merge. Thanks!

File: modules/deploying-lvms-on-sno-cluster.adoc (hunk 2 above)
Comment (sr1kar99) [reply modules/deploying-lvms-on-sno-cluster.adoc]: @bergerhoffer Thank you for pointing this out. I failed to notice this suggestion during the peer review. Updated the content based on the suggestion. PTAL.
Thanks.

**Review by bergerhoffer — COMMENTED** [review]
LGTM for merging!

**Comment by bergerhoffer** [comment]
@sr1kar99 it looks like versions 4.12-4.14 failed to pick cleanly. Can you please submit manual cherry picks for these versions?

I will merge the 4.15 version, which picked cleanly.
